torch/_dynamo/utils.py

```diff
@@ -219,185 +207,111 @@ def print_time_report() -> None:
+# Use the following singleton to capture and log CompilationMetrics. Entering the context
+# manager allocates a new record to be logged when it exits. (You should not need to use
+# this directly unless you introduce a new code path where compilation metrics would be
+# gathered). While compiling, use the setters or timer in MetricsContext to update fields
+# in the current context. For example:
+#
+# To set a single field:
+#   METRICS_CONTEXT.set("metric_name", value)
+#
+# To set multiple fields:
+#   METRICS_CONTEXT.update({"name1": val1, "name2": val2})
+#
+# To increment a numeric field:
+#   METRICS_CONTEXT.increment("metric_name", value)
+#
+# To record execution time, METRICS_CONTEXT works with dynamo_timed:
+#    def foo(...):
+#        with dynamo_timed("metric", dynamo_compile_column="metric_us")
+#            ...
+#
+METRICS_CONTEXT: MetricsContext
 
 
 # dynamo_timed is a context manager
-# By wrapping a function in dynamo_timed, we can store a record in compilation_time_metrics
-# where the key is the functions name.
-# For example:
+# By wrapping a function in dynamo_timed, we can get a few things:
+#
+# 1) Logging timings to pt2_compile_events.
+# 2) Logging timings to CompilationMetrics (including dynamo_compile).
+# 3) Chromium events.
+# 4) Storing a record in compilation_time_metrics
+#    For example:
 #
-#   def _foo(...):
-#       with dynamo_timed("_foo"):
-#           ...
+#     def _foo(...):
+#         with dynamo_timed("_foo"):
+#             ...
 #
-# Would show up as an entry in our timing dict:
-# OrderedDict([('_foo', [0.083690, 0.23949, 3.1425e-05])])
-# This is extremely useful for granular debugging.
+#     Would show up as an entry in our timing dict:
+#     OrderedDict([('_foo', [0.083690, 0.23949, 3.1425e-05])])
+#     This is extremely useful for granular debugging.
 #
 # Although it is tempting to use dynamo_timed as a decorator, please do not.
 # In its decorator form it makes cProfile traces less useful as dynamo_timed
 # suddenly becomes a bottleneck for lots of function calls (as only one parent
 # pointer is recorded).
-#
-# For a higher-level mode, pass a phase_name into dynamo_timed
-# phase_names record an extra record into a separate compilation timing structure,
-# one keyed on frame+name rather than function.
-# The frame is incremented outside of this function, in def increment_frame() above.
-# `fwd_only` is used to identify if this phase or function is only called
-# during compiling fwd graphs, e.g, `entire_frame_compile` and `backend_compile`.
-# The other phases (`inductor_compile` and `code_gen`) are called for both fwd and bwd graphs.
-
-
 @contextmanager
 def dynamo_timed(
-    key: str,
-    phase_name: Optional[str] = None,
-    log_pt2_compile_event: bool = False,  # Whether or not to log it to internal pt2 compile event
-    fwd_only: bool = True,
+    event_name: str,
+    # TODO: do we really want event and fn name?
+    fn_name: Optional[str] = None,
```

Comment:
No

Reply:
Like, I accept you may need to keep it for BC / keeping tests happy, but in the terminal state I don't think it adds value.